CONTRIBUTING.md

```diff
@@ -0,0 +1,122 @@
+# Contributing to [Project-Name]
+
+## Table of Contents
+
+1. [Code Style](#code-style)
+2. [Documentation](#documentation)
+3. [Directory Structure](#directory-structure)
+4. [Testing](#testing)
+5. [Version Control](#version-control)
+6. [Code Reviews](#code-reviews)
+7. [Additional Resources](#additional-resources)
+
+## Code Style
+
+Please follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) guidelines for writing Python code.
+
+- **Indentation**: Use 4 spaces per indentation level.
+- **Line Length**: Limit all lines to a maximum of 79 characters.
+- **Naming Conventions**:
+  - Functions and variables: `snake_case` (e.g., `train_model`, `learning_rate`)
+  - Classes: `CamelCase` (e.g., `CNNClassifier`)
+- **Imports**: Group imports into three categories in the following order:
+  1. Standard library imports.
+  2. Related third-party imports.
+  3. Local application/library-specific imports.
+- **String Quotes**: Use single quotes for strings ('example') except when the string contains a single quote that needs escaping.
+- **Code Formatting**: Use `black` for automatic code formatting to ensure consistency and adherence to PEP 8 guidelines.
+
+## Documentation
+
+- **Docstrings**: Document all public modules, functions, classes, and methods.
+- **Format**: Use Google style docstrings.
+
+### Example:
+
+```python
+def train_model(data, labels, epochs):
+    """
+    Trains the CNN model on the provided data and labels.
+
+    Args:
+        data (np.ndarray): Training data.
+        labels (np.ndarray): Training labels.
+        epochs (int): Number of epochs to train.
+
+    Returns:
+        model: Trained CNN model.
+    """
+    pass
+```
+
+## Directory Structure
+
+```
+Project-Name/
+├── scripts/
+│   ├── main.py
+│   ├── preprocess_data.py
+├── src/
+│   ├── data.py
+│   ├── model.py
+├── tests/
+│   ├── test_load_data.py
+│   ├── test_preprocess_data.py
+├── notebooks/
+│   └── exploratory_data_analysis.ipynb
+├── .github/
+│   └── pull_request_template.md
+├── README.md
+├── CONTRIBUTING.md
+├── requirements.txt
+├── pre-commit-conflig.yaml

```

Comment:
We fetched the branch and installed Python 11, installed hooks from the pre-commit-config.yaml, and created a sample file. Then we made a commit, but pre-commit didn't work.

Reply:
I've checked it, the file's name was wrong (pre-commit-conflig.yaml instead of .pre-commit-conflig.yaml). I fixed it and it works for me, please check if it works for you. If needed the instruction is in README.md file